packages/snap/src/index.ts

```diff
@@ -53,46 +61,27 @@ export const processRpcRequest: SnapRpcProcessor = async (
     case RpcMethods.GenZkKycProof: {
       // parse ZKP inputs
       const genParams = request.params as unknown as GenZkProofParams<any>;
-      // check some input validity
-      if (genParams.userAddress === undefined) {
-        throw new GenZKPError({
-          name: 'MissingInputParams',
-          message: `userAddress missing in request parameters.`,
-        });
-      }
-      if (genParams.requirements.zkCertStandard === undefined) {
-        throw new GenZKPError({
-          name: 'MissingInputParams',
-          message: `ZkCert standard missing in request parameters.`,
-        });
-      }
+      checkZkKycProofRequest(genParams);
 
       const zkCert = await selectZkCert(
         snap,
         state.zkCerts,
         genParams.requirements.zkCertStandard,
+        genParams.requirements.registryAddress,
       );
+      holder = getHolder(zkCert.holderCommitment, state.holders);
 
-      const searchedHolder = state.holders.find(
-        (candidate) => candidate.holderCommitment === zkCert.holderCommitment,
-      );
-      if (searchedHolder === undefined) {
-        throw new GenericError({
-          name: 'MissingHolder',
-          message: `Holder for commitment ${zkCert.holderCommitment} could not be found. Please use Metamask with the same mnemonic as when you created this holder commitment.`,
-        });
-      }
-      holder = searchedHolder;
+      const searchedZkCert = getZkCert(zkCert.leafHash, state.zkCerts);
 
-      const searchedZkCert = state.zkCerts.find(
-        (cert) => cert.leafHash === zkCert.leafHash,
+      const merkleProof = await getMerkleProof(
```

Comment:
Since we encrypt the zkCert from KYC provider back to the user to be secure, even though the merkle proof data are much less important than the zkCert data, we might also need to encrypt the returned MerkleProof to prevent account linking, if we want to be fully careful.

Reply:
I see your point.
The Merkle proof here is stored in the Metamask storage and should be sent using https. Both is encrypted. (Added a check for the fetch URL to use https.)
Therefore I think that we do not need to encrypt it again.
This is different for zkCerts because it is passed as file that might be sent using EMail or put into a download folder on the file system.